src/Pages/ProductEdit.tsx

```diff
@@ -0,0 +1,274 @@
+import React, { useEffect, useState } from "react";
+import { useForm } from "react-hook-form";
+import { object, string, number } from "yup";
+import { yupResolver } from "@hookform/resolvers/yup";
+import {
+  Box,
+  Button,
+  Card,
+  CardContent,
+  Divider,
+  TextField,
+  Typography,
+  MenuItem,
+  styled,
+} from "@mui/material";
+import axios from "axios";
+import { useNavigate, useParams } from "react-router-dom";
+import { TerpenEnum } from "../types/Terpen";
+import { GeneticsEnum } from "../types/GeneticsEnum";
+import { useToken } from "../hooks/useToken";
+import { Spinner } from "../components/Spinner";
+
+const productSchema = object({
+  producentName: string().min(4).max(20).required("This field is required"),
+  strainName: string().min(4).max(20).required("This field is required"),
+  genetics: string().min(4).max(15).required(),
+  terpen: string().required(),
+  thcLevel: number().positive().integer().min(1).required(),
+  cbdLevel: number().positive().integer().min(1).required(),
+  description: string().required(),
```

Comment:
przy `.required()` powinien byc jakis error message, tak jak w `producentName` i `strainName`; biorac tez pod uwage to, ze uzywasz ich potem w TextFieldach. Obecnie nigdy zaden error message sie nie pojawi, bo nie istnieje